kie-soup-dataset/kie-soup-dataset-sql/src/test/java/org/dashbuilder/dataprovider/sql/SQLDataSetProviderTest.java

```diff
@@ -0,0 +1,97 @@
+
+import static org.hamcrest.CoreMatchers.endsWith;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.dashbuilder.dataprovider.sql.model.Column;
+import org.dashbuilder.dataset.ColumnType;
+import org.dashbuilder.dataset.DataSetMetadata;
+import org.dashbuilder.dataset.def.DataSetDefFactory;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({JDBCUtils.class})
+public class SQLDataSetProviderTest {
+
+    @Mock
+    Statement statement;
+    
+    @Mock
+    ResultSet resultSet;
+    
+    @Spy
+    SQLDataSetProvider dataSetProvider = SQLDataSetProvider.get();
+    
+    ResultSetHandler resultSetHandler;
+    
+    @Before
+    public void setUp() throws Exception {
+        PowerMockito.mockStatic(JDBCUtils.class);
+        
+        final List<Column> dbColumns = new ArrayList<Column>();
+        dbColumns.add(SQLFactory.column("dbWins1", ColumnType.LABEL, 10));
+        dbColumns.add(SQLFactory.column("dbWins2", ColumnType.LABEL, 10));
+        dbColumns.add(SQLFactory.column("dbWins3", ColumnType.LABEL, 10));
+
+        dataSetProvider.setDataSourceLocator(new DatabaseTestSettings().getDataSourceLocator());
+        resultSetHandler = new ResultSetHandler(resultSet, statement);
+        
+        when(JDBCUtils.dialect(any(Connection.class))).thenReturn(JDBCUtils.H2);
+        when(JDBCUtils.executeQuery(any(Connection.class), any())).thenReturn(resultSetHandler);
+        when(JDBCUtils.getColumns(any(),any())).thenReturn(dbColumns);
+        when(JDBCUtils.fixCase(any(Connection.class),eq("test"))).thenReturn("TEST");
+    }
+
+    @Test
+    public void testGetColumnsWithLimitZero() throws Exception {
+        DataSetMetadata metadata = dataSetProvider.getDataSetMetadata(
+                DataSetDefFactory.newSQLDataSetDef()
+                        .uuid("test")
+                        .estimateSize(false)
+                        .dbTable("test", true)
+                        .buildDef());
+
+        ArgumentCaptor<Connection> conn = ArgumentCaptor.forClass(Connection.class);
+        ArgumentCaptor<String> sql = ArgumentCaptor.forClass(String.class);
+
+        PowerMockito.verifyStatic(JDBCUtils.class);
+        JDBCUtils.executeQuery(conn.capture(), sql.capture());
+        assertThat(sql.getValue().toUpperCase(), endsWith("LIMIT 0"));
```

Comment:
There are some dbs that do not support LIMIT = 0, for example Oracle before 11g use ROWNUM <=0. . As far as we are aware that this is test is valid just for H2, for me is fine.

Reply:
Good point @fjtirado, thanks!
Added parameterized tests for all dialects, and also your suggestion for Sybase _getSelectStatement_ fixed.
Please review to check that it's correct.